pkg/vm/manager.go

```diff
@@ -0,0 +1,404 @@
+package vm
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+	"strconv"
+	"syscall"
+	"time"
+
+	firecracker "github.com/firecracker-microvm/firecracker-go-sdk"
+	"github.com/firecracker-microvm/firecracker-go-sdk/client/models"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/zos/pkg"
+)
+
+const (
+	// FCBin  path for firecracker
+	FCBin = "/bin/firecracker"
+	// JailerBin path for fc jailer
+	JailerBin = "/bin/jailer"
+	// FCSockDir where vm firecracker sockets are kept
+	FCSockDir = "/var/run/firecracker"
+
+	defaultKernelArgs = "ro console=ttyS0 noapic reboot=k panic=1 pci=off nomodules"
+)
+
+// vmModuleImpl implements the VMModule interface
+type vmModuleImpl struct {
+	root string
+}
+
+var (
+	_ pkg.VMModule = (*vmModuleImpl)(nil)
+
+	errScanFound = fmt.Errorf("found")
+)
+
+// NewVMModule creates a new instance of vm manager
+func NewVMModule(root string) (pkg.VMModule, error) {
+	if err := os.MkdirAll(FCSockDir, 0755); err != nil {
+		return nil, err
+	}
+
+	return &vmModuleImpl{
+		root: root,
+	}, nil
+}
+
+func (m *vmModuleImpl) makeDisk(name string, size int64) error {
```

Comment:
This needs to be moved into `storaged`

Reply:
Yup, we will do that, I just need to finalize few things in vmd (trying to run the k3s image) before I move to storaged

Reply:
should be deleted now ?

Reply:
yeah, it's not used anymore, will remove the dead code